Index: wiringPi-b0a60c3302973ca1878d149d61f2f612c8f27fac/Makefile
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ wiringPi-b0a60c3302973ca1878d149d61f2f612c8f27fac/Makefile	2016-04-11 21:35:24.071161242 +0800
@@ -0,0 +1,11 @@
+sudo="sudo"                                                                                                                                                                                                          
+
+default:
+	  cd wiringPi;make -j5;
+
+xinstall:
+	  cd wiringPi; $(sudo) make install
+		  
+xuninstall:
+	  cd wiringPi; $(sudo) make uninstall
+
